src/ralph/api/routers/statements.py

```diff
@@ -254,3 +255,77 @@ async def get(
             statement["_source"] for statement in es_response["hits"]["hits"]
         ],
     }
+
+
+@router.post(
+    "/",
+    responses={
+        400: {
+            "model": ErrorDetail,
+            "description": "The request was invalid.",
+        },
+        409: {
+            "model": ErrorDetail,
+            "description": "Statements had a conflict with existing statements.",
+        },
+    },
+)
+# pylint: disable=unused-argument
+async def post(
+    statements: List[LaxStatement],
+):
+    """
+    Store a set of statements (or a single statement as a single member of a set).
+    NB: at this time, POST requests passing a statement directly as the body of the
+    request, or using POST to make a GET request, are not supported.
+
+    LRS Specification:
+    https://github.com/adlnet/xAPI-Spec/blob/1.0.3/xAPI-Communication.md#212-post-statements
+    """
+
+    # The statements dict has multiple functions:
+    # - generate IDs for statements that are missing them;
+    # - use the list of keys to perform validations and as a final return value;
+    # - provide an iterable containing both the statements and generated IDs for bulk.
+    statements_dict = {
+        str(statement.id) if statement.id else str(uuid4()): statement
+        for statement in statements
+    }
+
+    # Requests with duplicate statement IDs are considered invalid
+    # statements_ids were deduplicated by the dict, statements list was not
+    statements_ids = list(statements_dict.keys())
+    if len(statements) != len(statements_ids):
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Duplicate statement IDs in the list of statements",
+        )
+
+    es_response = ES_CLIENT.search(
+        index="statements", body={"query": {"terms": {"_id": statements_ids}}}
+    )
+
+    if len(es_response["hits"]["hits"]) > 0:
+        # NB: LRS specification calls for performing a deep comparison of incoming
+        # statements and existing statements with the same ID.
+        # This seems too costly for performance and was not implemented for this POC.
+        raise HTTPException(
+            status_code=status.HTTP_409_CONFLICT,
+            detail="Statements already exist with the same ID",
+        )
+
+    # For valid requests, perform the bulk indexing of all incoming statements
+    bulk(
+        ES_CLIENT,
+        [
+            {
+                "_index": "statements",
+                "_id": statement_id,
+                "_op_type": "index",
+                "_source": statement.json(),
+            }
+            for statement_id, statement in statements_dict.items()
+        ],
+    )
```

Comment:
Maybe we should provide more details to ease debug, WDYT?

```suggestion
    try:
        actions, errors = bulk(
            ES_CLIENT,
            [
                {
                    "_index": "statements",
                    "_id": statement_id,
                    "_op_type": "index",
                    "_source": statement.json(),
                }
                for statement_id, statement in statements_dict.items()
            ],
        )
    except BulkIndexError:
        logger.error("Failed to index submitted statements")
        logger.debug("Error was: %s", str(errors))
        raise HTTPException(status_code=500, detail="Statements bulk indexation failed")
        
    logger.info("Indexed %d statements with success", actions)
```